### PR TITLE
invenio: Add option to enable/disable podSecurityContext

### PR DESCRIPTION
**Pull request by mb-wali**
:heart: Thank you for your contribution!

### Description

> Please describe briefly your pull request.
In OpenShift, where the user ID (UID) and group ID (GID) are automatically managed, the ability to remove these settings from the pod security context simplifies deployment

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [ ] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [ ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).
    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Frontend**

- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines.
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines.
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines.

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

**Comment by ntarocco** [comment]
@lindhe @Samk13 can you please have a look?

**Comment by lindhe** [comment]
@ntarocco I can take it. Don't think I can add myself as reviewer or assignee, but you can consider me as such.

File: charts/invenio/templates/web-deployment.yaml [hunk 1]
```diff
@@ -214,8 +214,8 @@ spec:
               cpu: '1'
               memory: 100Mi
             requests:
-              cpu: '1'
-              memory: 100Mi
+              cpu: 500m
+              memory: 50Mi
```
Comment (lindhe) [inline charts/invenio/templates/web-deployment.yaml:218]: These changes seem unrelated to the PR. Please revert them.

**Review by lindhe — CHANGES_REQUESTED** [review]
I like the intention of this PR – giving the users the option to enable or disable the security context is good. And I think your changes look good; they do what you intend them to do (as far as I can tell). Just need to restore the resource requests that accidentally got included, according to my comment.

However, I think it would be even more useful if we gave the users the ability to fully configure the security contexts rather than just an on/off switch. Preferably configurable for each pod and container, similar to [how Bitnami does it](https://github.com/bitnami/charts/blob/0868c633cc6b2f16efcbbd86e0dce126ac0fdcd1/bitnami/postgresql/values.yaml#L481-L520). Is it clear to you how I mean?

Would you be willing to rewrite the values in such a manner instead? I think that would be more valuable to this project. If that's too much work, I think it can be retrofitted later when the need arises.

File: charts/invenio/templates/web-deployment.yaml (hunk 1 above)
Comment (mb-wali) [reply charts/invenio/templates/web-deployment.yaml:218]: You're absolutely correct; these changes are unrelated to the PR. However, I noticed that the current resource requests seemed a bit high for an init container, so I made a small adjustment to optimize it. Later, it might be helpful to define this in the values, just like the parent container resources. Let me know if you'd still like me to revert these changes.

**Comment by mb-wali** [comment]
> I like the intention of this PR – giving the users the option to enable or disable the security context is good. And I think your changes look good; they do what you intend them to do (as far as I can tell). Just need to restore the resource requests that accidentally got included, according to my comment.
> 
> However, I think it would be even more useful if we gave the users the ability to fully configure the security contexts rather than just an on/off switch. Preferably configurable for each pod and container, similar to [how Bitnami does it](https://github.com/bitnami/charts/blob/0868c633cc6b2f16efcbbd86e0dce126ac0fdcd1/bitnami/postgresql/values.yaml#L481-L520). Is it clear to you how I mean?
> 
> Would you be willing to rewrite the values in such a manner instead? I think that would be more valuable to this project. If that's too much work, I think it can be retrofitted later when the need arises.

Certainly, I can see the value in making the security context fully configurable, as you suggested, similar to Bitnami’s approach. I'd be happy to proceed with the current scope of the PR for now and then open an issue to address a more comprehensive configuration in the future. This way, we can implement the core changes immediately and expand on them later when the need arises. Thanks for the thoughtful feedback!

File: charts/invenio/templates/web-deployment.yaml (hunk 1 above)
Comment (lindhe) [reply charts/invenio/templates/web-deployment.yaml:218]: I agree that they can most likely be improved. But that improvement belongs in a separate PR. 👍
